index.js

```diff
@@ -144,5 +146,13 @@ module.exports = function Addon(manifest) {
 		publishToDir(baseDir || './publish-'+manifest.id, manifest, handlers)
 	}
 
+	this.publishToWeb = function(addonUrl) {
+		if (!addonUrl) throw 'Please set a valid url'
+
+		// Create a protocol url
+		manifestRender.protocol = addonUrl.replace(/^(https?|ftp):\/\//, 'stremio://') + '/manifest.json';
```

Comment:
why are we cloning the manifest and adding the manifest url to the cloned manifest? we could simply use a separate variable for it so it leads to cleaner code altogether

also, keep in mind that `stremio://` links only work with `https`, so changing `http` and `ftp` protocols to `stremio://` too won't help, as those won't work in Stremio

Reply:
Cloning is to avoid modifying the manifest itself and to avoid that the returned manifest is different from the one created by the user
The unit test specify that : ` // NOTE: this is an important design principle - immutability on the manifest object`

So I specified that it must be an https url by throwing an error

Reply:
yes, this is true, the manifest should be immutable, and it should not be cloned to be modified either, what i'm proposing is that we don't add the developer defined url at all to the manifest object, we just keep it in a separate variable that we make available in the web page

we don't need to add it to the manifest at all, not even a cloned version of it, we just need +1 variable available in the web page that stores the manifest's url.

Reply:
Oh okay i see, but i don't understand what i'm supposed to do.
The manifestRender variable was for rendering purpose only, just for filling the handlebars template with all the manifest data + protocol and logo/background. It was not supposed to be returned to the user in other ways.

Do you mean like this ?
```
// Set default logo & background if not set in the manifest
const logo = manifest.logo || `/static/imgs/logo.png`;
const background = manifest.background || `/static/imgs/background.jpg`;
const protocol = null;

...
// Create a protocol url
protocol = addonUrl.replace('https://', 'stremio://') + '/manifest.json';
...

// Render the home page
addonHTTP.get('/', function (req, res) {
   res.render('home', { manifest: manifest, logo: logo, background: background, protocol: protocol});
});
```

Reply:
exactly, but because we're using ES6, you can simplify that code, by doing:

```
res.render('home', { manifest, logo, background, protocol });
```

and the properties will still have the same name

can we also change the variable name from `protocol` to `manifestUrl`?

as it's not really a protocol, and it can get confusing in the code if we call it that

Reply:
Oh ok, thanks, i'll do that !

Reply:
ok, looks good, will just go over the code one more time and test it again